src/App.tsx

```diff
@@ -1,21 +1,71 @@
-import './App.css'
 import CarForm from './components/carForm'
 import { useState } from 'react'
 import { Car } from './types'
+import VCV from '../lib/vcv'
 import CarList from './components/carList'
+import CarEditDialog from './components/carEditDialog'
+import CarHistory from './components/carsHistory'
+import { Box, Button, Container, Divider, ThemeProvider} from '@mui/material'
+
+import themes from './themes'
 
 function App() {
-  const [cars, setCars] = useState<Car[]>([]) // Car array state
+	const { state : cars, 
+					setValue : setCars, 
+					revertTo : revertCommit, 
+					history : commitHistory,
+					index: commitIndex } = VCV<Car[]>([]) // Car array state
+	const [themeIndex, setThemeIndex] = useState(0)
+	const [editDialogOpen, setEditDialogOpen] = useState(false)
+	const [editIndex, setEditIndex] = useState(-1)
+
+	function changeTheme() {
+		setThemeIndex((themeIndex + 1) % themes.length)
+	}
+
+	function addCar(car: Car) { // Function to add a car to the list
+		setCars([...cars, car])
+	}
+
+	function deleteCar(i: number) { // Function to delete a car from the list
+		setCars(cars.filter((_, index) => index !== i))
+	}
+	
+	function editCar(i: number, car: Car) { // Function to edit a car from the list
+		setCars(cars.map((c, index) => index === i ? car : c))
+	}
 
-  
+	function handleEdit(i: number) { // Function to handle car edition
+		setEditIndex(i)
+		setEditDialogOpen(true)
+	}
 
-  return (
-    <div className='cars-container'>
-      <CarForm cars={cars} setCars={setCars} />
-      <div className='divider'/>
-      <CarList cars={cars} setCars={setCars} />
-    </div>
-  )
+	return (
+		<ThemeProvider theme={themes[themeIndex].theme} >
+			<Box 
+				sx={{height: '100vh', width: '100vw', alignContent: 'center', justifyContent: "center", bgcolor: 'background.default'}}
+				className='cars-container'>
+					<Container sx={{display: 'flex', maxHeight: '80vh', flexDirection: {xs: "column", md: "row"}, justifyContent: 'center', alignItems: "center"}}>
+						<CarForm addCar={addCar} />
+						{cars.length !== 0 && <Divider orientation='vertical' flexItem variant='middle' sx={{m: 2}} />}
+						<Box sx={{display: 'flex', flexDirection: 'column', alignItems: 'end'}}>
+							<CarHistory 
+								currentIndex={commitIndex}
+								history={commitHistory} 
+								revertTo={revertCommit}/>
+							<CarList cars={cars} deleteCar={deleteCar} editCar={handleEdit} />
+						</Box>
+					</Container>
+					<Button sx={{position: 'absolute', top: 5, right: 5}} onClick={changeTheme} variant='text'>{themes[themeIndex].icon}</Button>
```

Comment:
If you don't have any text in a button and the icon only, it's better to use MUI `<IconButton>` component instead.
Also, if you would like to add some text, you can keep <Button> component and pass icon as `startIcon` or `endIcon` prop